pool.go

```diff
@@ -61,9 +71,49 @@ func (p *pool) Store(v wire) {
 func (p *pool) Close() {
 	p.cond.L.Lock()
 	p.down = true
+	p.stopTimer()
 	for _, w := range p.list {
 		w.Close()
 	}
 	p.cond.L.Unlock()
 	p.cond.Broadcast()
 }
+
+func (p *pool) startTimerIfNeeded() {
+	if p.idleConnTTL == 0 || p.timerIsActive || len(p.list) <= p.minSize {
+		return
+	}
+
+	p.timerIsActive = true
+	if p.timer == nil {
+		p.timer = time.AfterFunc(p.idleConnTTL, p.removeIdleConns)
+		return
+	}
+
+	p.timer.Reset(p.idleConnTTL)
+}
+
+func (p *pool) removeIdleConns() {
+	p.cond.L.Lock()
+	defer p.cond.L.Unlock()
+
+	if p.down || p.size <= p.minSize || len(p.list) == 0 {
+		return
+	}
+
+	newLen := min(p.minSize, len(p.list))
+	for _, w := range p.list[newLen:] {
+		w.Close()
+	}
+
+	p.list = p.list[:newLen]
+	p.size = p.minSize
```

Comment:
```suggestion
```

Setting it directly to the `p.minSize` seems incorrect. We probably should put it in the above loop and use `p.size--` instead.